can-io-firmware/src/pwm.h

```diff
@@ -0,0 +1,27 @@
+#ifndef PWM_H
+#define PWM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <ch.h>
+#include <hal.h>
+
+enum pwm_channel
+{
+    PWM_CHANNEL_0, // Timer 16 channel 0
+    PWM_CHANNEL_1, // Timer 17 channel 0
+    PWM_CHANNEL_2, // Timer 1 channel 1
+    PWM_CHANNEL_3, // Timer 1 channel 2
+};
+
+void pwm_init(uint32_t frequency, uint32_t period);
+
+void pwm_set_pulsewidth(enum pwm_channel channel, uint32_t pulsewidth);
```

Comment:
Maybe add a comment / variable name to give the units of period and pulsewidth ? Is it ms or clock ticks ?

Reply:
Clock ticks, adding a comment